File: 3280.convert-date-to-binary.py
#
# @lc app=leetcode id=3280 lang=python
#
# [3280] Convert Date to Binary
#

# @lc code=start
class Solution(object):
    def convertDateToBinary(self, date):
        """
        :type date: str
        :rtype: str
        """
        output = ''
        terms = date.split('-')
        for i in terms:
            output+= bin(int(i))[2::] 
            output += '-'
        return output[:-1:]
        
        
# @lc code=end

File: 3280.convert-date-to-binary.java
/*
 * @lc app=leetcode id=3280 lang=java
 *
 * [3280] Convert Date to Binary
 */

// @lc code=start
class Solution {
    public String convertDateToBinary(String date) {
        StringBuilder sb = new StringBuilder("");
        String year = Integer.toBinaryString(Integer.parseInt(date.substring(0,4)));
        String month = Integer.toBinaryString(Integer.parseInt(date.substring(5,7)));
        String dat = Integer.toBinaryString(Integer.parseInt(date.substring(8,10)));
        sb.append(year);
        sb.append("-");
        sb.append(month);
        sb.append("-");
        sb.append(dat);
        return sb.toString();
    }
}
// @lc code=end

